x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/agent_policy_advanced_fields/custom_fields/global_data_tags_table.tsx

```diff
@@ -0,0 +1,391 @@
+
+    if (!isValid) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        [index]: { name: nameError, value: valueError },
+      }));
+      return;
+    }
+
+    const updatedTags = globalDataTags.map((t, i) =>
+      i === index
+        ? { ...tag, name: tag.name.trim(), value: parseValue(tag.value.toString().trim()) }
+        : t
+    );
+    setGlobalDataTags(updatedTags);
+    updateAgentPolicy({ global_data_tags: updatedTags });
+    setEditIndexList((prevIndices: Set<number>) => {
+      const newIndices = new Set(prevIndices);
+      newIndices.delete(index);
+      return newIndices;
+    });
+    setErrors((prevErrors) => {
+      const newErrors = { ...prevErrors };
+      delete newErrors[index];
+      return newErrors;
+    });
+  };
+
+  const handleStartEdit = (index: number) => {
+    setEditIndexList((prevIndices) => new Set(prevIndices.add(index)));
+    setErrors((prevErrors: Record<number, { name: string | null; value: string | null }>) => ({
+      ...prevErrors,
+      [index]: { name: null, value: null },
+    }));
+  };
+
+  const cancelNewTagChanges = () => {
+    setNewTag({ name: '', value: '' });
+    setIsAdding(false);
+    setNewTagErrors({ name: '', value: '' });
+  };
+
+  const cancelEditChanges = (index: number) => {
+    setEditIndexList((prevIndices) => {
+      const newIndices = new Set(prevIndices);
+      newIndices.delete(index);
+      return newIndices;
+    });
+    setErrors((prevErrors) => {
+      const newErrors = { ...prevErrors };
+      delete newErrors[index];
+      return newErrors;
+    });
+  };
+
+  const showModal = (index: number) => {
+    setDeleteIndex(index);
+    setIsModalVisible(true);
+  };
+
+  const closeModal = () => {
+    setIsModalVisible(false);
+    setDeleteIndex(null);
+  };
+
+  const handleDelete = (index: number) => {
+    const updatedTags = globalDataTags.filter((_, i) => i !== index);
+    setGlobalDataTags(updatedTags);
+    updateAgentPolicy({ global_data_tags: updatedTags });
+    setEditIndexList((prevIndices: Set<number>) => {
+      const newIndices = new Set(prevIndices);
+      newIndices.delete(index);
+      return newIndices;
+    });
+    setErrors((prevErrors) => {
+      const { [index]: removedError, ...remainingErrors } = prevErrors;
+      return remainingErrors;
+    });
+    closeModal();
+  };
+
+  const deleteConfirmModal = (
+    <EuiOverlayMask>
+      <EuiConfirmModal
+        title={`Remove the ${deleteIndex ? globalDataTags[deleteIndex].name : ''} field?`}
+        onCancel={closeModal}
+        onConfirm={() => handleDelete(deleteIndex!)}
+        cancelButtonText="Cancel"
+        confirmButtonText="Remove"
+        buttonColor="danger"
+        defaultFocusedButton="confirm"
+      >
+        <p>Removing the field will affect the next sync. This action cannot be undone.</p>
+      </EuiConfirmModal>
+    </EuiOverlayMask>
+  );
+
+  const columns = [
```

Comment:
this variable needs to have a type:

```suggestion
    const columns: Array<EuiBasicTableColumn<GlobalDataTag>> = [
```

you can import it with 
```
import { EuiBasicTableColumn } from '@elastic/eui';